### PR TITLE
Add CommonJS support and external TypeScript definitions.

### PR DESCRIPTION
**Pull request by jnetterf**
This allows projects in environments such as Node, Webpack, Browserify, and React Native to use WebCola. In addition, TypeScript projects will be able to add WebCola type definitions to their projects by running `tsd link` (requires tsd 0.6).

The goal of this PR is to be as **non-invasive** as possible. This PR adds CommonJS support with a single-line change to existing code (`rbtree.js`). If you prefer external modules, you could revisit [this strategy](https://github.com/tgdwyer/WebCola/pull/102) later, but it seems to me that internal modules are working quite well in this project.

### Test Plan:

I tested WebCola in one of my own projects by adding

    "dependencies": {
      "webcola": "jnetterf/WebCola#commonjs"
    }

to `package.json`, running `npm link` to link to my local copy of WebCola, and running `tsd link` to register WebCola's type definitions.

I then opened up a file in Visual Studio Code for Mac, and tested the simplest thing I could think of:

    import Cola = require("webcola");
    let a = new Cola.vpsc.Rectangle(0, 20, 0, 20);
    let b = new Cola.vpsc.Rectangle(0.01, 20, 0, 20);
    let c = [a,b];
    Cola.vpsc.removeOverlaps(c);
    console.log(c);

IntelliSense was available while writing the Hello World test, and the test program ran in Node.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ###############################################################################
 # Set default behavior to automatically normalize line endings.
 ###############################################################################
-* text=auto
+# * text=auto
```
Comment (jnetterf) [inline .gitattributes]: Line endings across platforms are a pain! LF line normalization was added in `8f5eaadba32b322`, but the lines weren't actually normalized, which causes all sorts of weirdness for non-Windows users. Would you like me to prepare a PR to normalize line endings?

File: WebCola/src/layout.ts [hunk 2]
```diff
@@ -37,7 +37,7 @@ module cola {
      */
     export class Layout {
         private _canvasSize = [1, 1];
-        private _linkDistance: number | ((any) => number) = 20;
```
Comment (jnetterf) [inline WebCola/src/layout.ts]: As written, `any` is an argument name, not a type.

Comment (tgdwyer) [inline WebCola/src/layout.ts]: good catch!

File: .gitattributes (hunk 1 above)
Comment (tgdwyer) [inline .gitattributes]: That would be great,  thanks!

File: WebCola/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,972 @@
+/// <reference path="extern/d3.d.ts" />
```
Comment (tgdwyer) [inline WebCola/index.d.ts]: Is this file auto-generated somehow?  It would be good to document how we keep this index.d.ts file up-to-date.

**Pull request merged**